api/src/pcapi/alembic/versions/20240111T195604_cdb9a9f92d2b_add_partial_index_on_offer_extradata_for_music_type.py

```diff
@@ -0,0 +1,39 @@
+"""
+Add partial index for musicType in Offer.extraData
+"""
+from alembic import op
+import sqlalchemy as sa
+
+from pcapi import settings
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "cdb9a9f92d2b"
+down_revision = "f0fbdcf9961d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.execute("COMMIT")
+    op.execute("SET SESSION statement_timeout='300s'")  # helm upgrade timeout
```

Comment:
Combien de temps prend la requête de création de cet index sur staging ?

Reply:
pcapi-staging> EXPLAIN ANALYZE SELECT DISTINCT offer.id AS id FROM offer WHERE (offer."jsonData" ->> 'musicType') IN ('800') LIMIT 100;
+----------------------------------------------------------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                                                                   |
|----------------------------------------------------------------------------------------------------------------------------------------------|
| Limit  (cost=0.57..79967.30 rows=100 width=8) (actual time=4.501..28.017 rows=100 loops=1)                                                   |
|   ->  Unique  (cost=0.57..444427136.14 rows=555765 width=8) (actual time=4.500..28.000 rows=100 loops=1)                                     |
|         ->  Index Scan using offer_pkey on offer  (cost=0.57..444425746.73 rows=555765 width=8) (actual time=4.497..27.971 rows=100 loops=1) |
|               Filter: (("jsonData" ->> 'musicType'::text) = '800'::text)                                                                     |
|               Rows Removed by Filter: 23137                                                                                                  |
| Planning Time: 0.166 ms                                                                                                                      |
| Execution Time: 28.057 ms                                                                                                                    |
+----------------------------------------------------------------------------------------------------------------------------------------------+

Reply:
Je répète la question, que je n'ai pas placée sur cette ligne par hasard : 
Combien de temps prend la requête de **création de cet index** sur staging ?

Reply:
pcapi-staging> CREATE INDEX offer_music_type_idx ON public.offer USING btree ((("jsonData" ->> 'musicType'::text))) WHERE ((("jsonData" ->> 'musicType'::text))) IS NOT NULL
CREATE INDEX
Time: 2281.291s (38 minutes 1 second), executed in: 2281.291s (38 minutes 1 second)

Reply:
utilisation de https://www.notion.so/Clarifier-les-pratiques-de-migration-de-BDD-5f8edeba57ed4a17b80c847a74def027#7ffdc1ea90bc4cca90f1903d9ba0f573